src/term.rs

```diff
@@ -14,6 +14,12 @@ pub enum Term {
 impl Term {
     // Get `Term` by parsing a string
     pub fn from_str(s: &str) -> Result<Self, Error> {
+        // this is a little weird, should we default to Nonterminal?
+        // Seems like empty would correspond to a terminal that represents
+        // the empty string.
```

Comment:
I agree with this comment. Not sure if you meant this as an actual question for me to be removed later or if the doubt is something you would like to keep in comments